### ISSUE TITLE
Error during preparing system

### ISSUE THREAD

**Issue by myungjinlee** [prody/ProDy#1077]
<img width="605" alt="error" src="https://user-images.githubusercontent.com/32939217/84436929-3c4fdb80-ac02-11ea-8ca9-30d1f4b19f0d.png">

I encountered this error message even before starting the simulation.
Could you please advise me to fix the issue? Thanks!

**Comment by jamesmkrieger** [comment]
Could you provide more information please? What did you enter into the form? It looks like it's having difficulty understanding the chain ID that you entered. It thinks you entered something then reads it as being basically nothing I think.

I have made a potential fix based on the idea that it could be a space. Please can you download the latest version from github and see if that fixes it?

Many thanks
James

**Comment by myungjinlee** [comment]
I appreciate your quick response. I will try with the latest version that you provided. 
As for my system, I have correct chain ID in the pdb files. I entered protein with glycan chains. I am not sure if glycan makes problems on this because I tried with just protein itself and got the same error. 

I will first try the latest version and see if it fix the issues.

Thanks!
Myungjin

**Comment by jamesmkrieger** [comment]
Hi Myungjin,

Hopefully, it fixes the issue but I'm not sure if it will. I'm still not sure enough about what you did. Could you try and explain it again please and also what you mean by "correct chain IDs"?

As for glycans, I think they should be fine as long as you specify chains and provide topology and parameters files for them. If you don't specify chain IDs then it selects the protein only but then you wouldn't have the error about chain IDs that you have.

Many thanks
James

**Comment by myungjinlee** [comment]
Hi James,

I am not sure where I can find and how to install the latest version. Did you mean the latest coMD.tcl file? 

The system that I am working on has two different conformational states. I want to see the transition pathway of theses two. The coMD handles this question right? I am also curious how long it will take if the system has around 40,000 atoms with all-atom simulation. The structure has 6 chains A,B,C,D,E,F and is trimer that is composed of three monomers (AB, CD, EF). The glycans are bonded on the surface of the proteins and they also have chain IDs associated with where they bonded. I entered the initial structure in the initial PDB section, and final structure in the final PDB in coMD in VMD(VMD-1.9.3.). When I click prepare PDB, it shows the captured error that I showed you. I can share the PDBs if you need more info of this, but not through this public channel, maybe by email?

I appreciate your help!

Best,
Myungjin

**Comment by jamesmkrieger** [comment]
Hi Myungjin,

Yes, you would need to download the latest coMD.tcl file from https://github.com/prody/coMD/ and put it in the plugins/noarch/tcl/comd directory of your VMD in place of the existing one. You may want to replace the anmmc.py file and well or even the whole directory and you can use the github program to do that and more easily keep it up to date. You may need to install a new version of VMD in a directory that you can modify in order to do this. 

Yes, CoMD is definitely meant to do that and a run for systems of that size usually takes a few hours on a machine with GPU. The PDB files shouldn't be a problem so you don't need to send them (although you are welcome to email them to kriegerj@pitt.edu if you would like) but instead it is a problem with how you are telling CoMD to use them or with the CoMD itself. Hopefully once you have the last CoMD the problem will be resolved. 

Best wishes
James

**Comment by myungjinlee** [comment]
<img width="600" alt="Screen Shot 2020-06-17 at 12 09 38 PM" src="https://user-images.githubusercontent.com/32939217/84922118-6c303080-b093-11ea-84e3-c239bf7c35a5.png">

Hi James,

I tried the latest version of coMD and now I got this error because I am working with MAC. "cat proc/cpuinfo" works only in Linux OS I guess. Is there any package that is compatible with MAC OSX? Otherwise, I will try to figure out to remotely connect to my Linux machine and try it. 

Best,
Myungjin

**Comment by jamesmkrieger** [comment]
Hi Myungjin,

If you set a number of cores in the GUI then it won't try to calculate it with proc/cpuinfo and hopefully then you won't have this problem. I'll look into an alternative too. 

Best wishes
James

**Comment by jamesmkrieger** [comment]
This should fix it now: https://github.com/prody/coMD/pull/68

Please try again.

**Comment by myungjinlee** [comment]
<img width="431" alt="Screen Shot 2020-06-20 at 4 26 31 PM" src="https://user-images.githubusercontent.com/32939217/85211061-7fe9ca00-b313-11ea-90c0-591aeef4a7fb.png">
<img width="927" alt="Screen Shot 2020-06-20 at 4 26 57 PM" src="https://user-images.githubusercontent.com/32939217/85211063-824c2400-b313-11ea-8c17-1af5abdb248e.png">


Hi James,

That works! I was able to run it after fixing the issue, but now I have another issue as you can see the attach figures. It was terminated at some point by itself and the output log file is zero byte, without any other output files.

Best,
Myungjin

**Comment by jamesmkrieger** [comment]
Hi Myungjin,

That's good to hear. 

I don't know what happened with that but I'd guess your computer couldn't support actually running the simulation. I'd suggest copying the files to a cluster and running the simulation there with `vmd -dispdev text -e run.tcl` in the submit script (changing `run.tcl` to whatever you called it).

Best wishes
James

**Comment by myungjinlee** [comment]
Hi James,

The thing is that the error comes not during simulation step but during preparing the system. 
So I do not have any output.tcl file to run the simulation with command line. 

I am more familiar with Linux and I believe so as coMD, but because of COVID-19, I only have limited access to Linux without VMD-GUI from home. I am working on to get full remote control of Linux so that I can try it with Linux VMD-GUI. 

I appreciate your very active response and help. Stay safe!

Best,
Myungjin

**Comment by myungjinlee** [comment]
Hi James,

I got another question. I finally made it run and got some output files to run simulation such as .tcl, walker1_ionized.pdb etc.

My system has protein + glycan, but none of pdbs (walker1_ionized.pdb, walter2_ionized.pdb) in the output folder does not have glycan in it. I entered glycan parameter along with protein parameter in minimization step. I did not specify any parameters in ionization step because it kept giving me some errors as I attached below
<img width="1115" alt="Screen Shot 2020-06-28 at 1 26 49 AM" src="https://user-images.githubusercontent.com/32939217/85938811-739ad980-b8de-11ea-8fae-21b44d208d61.png">

Could you please help me to include glycan in the simulation? 
I am sorry for many questions and issues. 

Best,
Myungjin

**Comment by jamesmkrieger** [comment]
Hi Myungjin,

Did this get fixed by my suggestion to use CHARMM-GUI instead? 

It would, of course, be good to support glycans better anyhow, but I wanted double check anyway.

Best wishes
James
